File: content/search/source/includes/index/field-types/embeddedDocuments/code-snippets/nodejs/stored-source-mltpl-conf.js
const { MongoClient } = require("mongodb");

// Replace the connection string with your MongoDB deployment's connection string
const uri =  "<connection-string>";

const client = new MongoClient(uri);

async function run() {
  try {
    const database = client.db("sample_training");
    const collection = database.collection("companies");

    // Create the Atlas Search index definition for the embeddedDocuments field
    const index = {
        name: "default",
        definition: {
          "mappings": { 
            "dynamic": false,
            "fields": { 
              "products": { 
                "type": "embeddedDocuments",
                "dynamic": true,
                "storedSource": true
              } 
            }
          },
          "storedSource": {
            "include": [ "_id", "name" ]
          }
        }
    }

    // Create the index
    const result = await collection.createSearchIndex(index);
    console.log("New index name: " + result);
  } finally {
    await client.close();
  }
}

run().catch(console.dir);

File: content/search/source/includes/index/field-types/embeddedDocuments/code-snippets/csharp/stored-source-mltpl-conf.cs
using MongoDB.Bson;
using MongoDB.Driver;

// Replace the connection string with your MongoDB deployment's connection string
var uri = "<connection-string>";

var client = new MongoClient(uri);

var db = client.GetDatabase("sample_training");
var collection = db.GetCollection<BsonDocument>("companies");

// Create the Atlas Search index definition for the embeddedDocuments field
var index =  new CreateSearchIndexModel(
  "default", new BsonDocument
  {
    { "mappings", new BsonDocument
      {
        { "dynamic", false },
        { "fields", new BsonDocument
          {
            { "products", new BsonDocument
              {
                { "type", "embeddedDocuments" },
                { "dynamic", true },
                { "storedSource", true }
              }
            }
          }
        }
      }
    },
    { "storedSource", new BsonDocument
      {
        { "include", new BsonArray { "_id", "name" } }
      }
    }
  });

var result = collection.SearchIndexes.CreateOne(index);
Console.WriteLine($"New index name: {result}");